doc/rtd/development/testing.rst

```diff
@@ -133,7 +133,69 @@ Test argument ordering
   * ``pytest.mark.parametrize``
   * ``mock.patch``
 
+.. _ubuntu_test_pre_release:
+
+Test pre-release cloud-init
+===========================
+
+After the cloud-init team creates an upstream release, cloud-init will
+be released in the -proposed APT repository for a
+:ref:`period of testing<sru_testing>`. Users are encouraged to test their
+workloads on this pending release so that bugs can be caught and fixed prior
+to becoming more broadly available via the -updates repository. This guide
+describes how to test the pre-release package on Ubuntu.
+
+Add the -proposed repository pocket
+-----------------------------------
+
+The -proposed repository pocket will contain the cloud-init package to be
+tested prior to release in the -updates pocket.
+
+.. code-block:: bash
+
+    echo "deb http://archive.ubuntu.com/ubuntu $(lsb_release -sc)-proposed main" >> /etc/apt/sources.list.d/proposed.list
+    apt update
+
+Install the pre-release cloud-init package
+------------------------------------------
+
+.. code-block:: bash
+
+    apt install cloud-init
+
+Test the package
+----------------
+
+Whatever workload you use cloud-init for in production is the best one
+to test. This ensures that you can discover and report any bugs that the
+cloud-init developers missed during testing before cloud-init gets
+released more broadly.
+
+If issues are found during testing, please file a `new cloud-init bug`_ and
```

Comment:
In the context of this page I can see that you mean "If issues are found during testing [of the package from proposed" but that may be less obvious if someone searches the docs for how to file a bug, finds this, and leaves a note in Matrix - is that the desired outcome of a Matrix ping for every bug?